lemminx-maven/src/main/java/org/eclipse/lemminx/maven/PluginValidator.java

```diff
@@ -22,20 +29,62 @@
 	private final MavenPluginManager pluginManager;
 	private final RepositorySystemSession repoSession;
 
-	public PluginValidator(MavenProjectCache cache, RepositorySystemSession repoSession, MavenPluginManager pluginManager) {
+	public PluginValidator(MavenProjectCache cache, RepositorySystemSession repoSession,
+			MavenPluginManager pluginManager) {
 		this.cache = cache;
 		this.repoSession = repoSession;
 		this.pluginManager = pluginManager;
 	}
+	
+	public boolean validatePluginResolution(DiagnosticRequest diagnosticRequest) {
+		try {
+			MavenPluginUtils.getContainingPluginDescriptor(diagnosticRequest, cache, repoSession,
+					pluginManager);
+		} catch (PluginResolutionException | PluginDescriptorParsingException | InvalidPluginDescriptorException e) {
+			//Add artifactId diagnostic
+			e.printStackTrace();
+			String errorMessage = e.getMessage();
+			DOMNode pluginNode = DOMUtils.findClosestParentNode(diagnosticRequest, "plugin");
+			Optional<DOMNode> artifactNode = pluginNode.getChildren().stream().filter(node -> node.getLocalName().equals("artifactId")).findAny();
+			artifactNode.ifPresent(node -> {
+				DiagnosticRequest artifactDiagnosticReq = new DiagnosticRequest(node,
+						diagnosticRequest.getXMLDocument(), diagnosticRequest.getDiagnostics());
+				Diagnostic unresolvablePlugin = new Diagnostic(artifactDiagnosticReq.getRange(),
```

Comment:
A diagnosticRequest should be able to generate a diagnostic, given an error message and severity. Refactor this.

Reply:
Refactor this in a (very near-)future PR.